kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/dsl/MultiDeleteable.java

```diff
@@ -16,11 +16,47 @@
 
 package io.fabric8.kubernetes.client.dsl;
 
+import java.util.Arrays;
 import java.util.List;
 
-public interface MultiDeleteable<T> {
+public interface MultiDeleteable<T> extends ListDeletable<T> {
```

Comment:
I'm not convinced that we should extend `ListDeletable` here. It feels that both interfaces serve different purposes: `ListDeletable` has an internal context whereas `MultiDeleteable` expects to get the resources passed as parameters.

Keeping these two separate doesn't harm their usage. Any implementing class can still implement both interfaces. IMHO, interfaces should be a collection of methods that serve the same purpose. That doesn't apply to these two methods in my opinion.